doc/glossary.rst

```diff
@@ -1508,45 +1508,27 @@ functions or non-estimator constructors.
-        may be configured to use a threaded numerical processor library (like
-        MKL; see :ref:`FAQ <faq_mkl_threading>`).
-
-        ``n_jobs`` is an int, specifying the maximum number of concurrently
-        running jobs.  If set to -1, all CPUs are used. If 1 is given, no
-        joblib level parallelism is used at all, which is useful for
-        debugging. Even with ``n_jobs = 1``, parallelism may occur due to
-        numerical processing libraries (see :ref:`FAQ <faq_mkl_threading>`).
-        For n_jobs below -1, (n_cpus + 1 + n_jobs) are used. Thus for
-        ``n_jobs = -2``, all CPUs but one are used.
-
-        ``n_jobs=None`` means *unset*; it will generally be interpreted as
-        ``n_jobs=1``, unless the current :class:`joblib.Parallel` backend
-        context specifies otherwise.
-
-        The use of ``n_jobs``-based parallelism in estimators varies:
-
-        * Most often parallelism happens in :term:`fitting <fit>`, but
-          sometimes parallelism happens in prediction (e.g. in random forests).
-        * Some parallelism uses a multi-threading backend by default, some
-          a multi-processing backend.  It is possible to override the default
-          backend by using :func:`sklearn.utils.parallel_backend`.
-        * Whether parallel processing is helpful at improving runtime depends
-          on many factors, and it's usually a good idea to experiment rather
-          than assuming that increasing the number of jobs is always a good
-          thing.  *It can be highly detrimental to performance to run multiple
-          copies of some estimators or functions in parallel.*
-
-        Nested uses of ``n_jobs``-based parallelism with the same backend will
-        result in an exception.
-        So ``GridSearchCV(OneVsRestClassifier(SVC(), n_jobs=2), n_jobs=2)``
-        won't work.
+        This parameter is used to specify how many concurrent processes or
+        threads should be used for routines that are parallelized with
+        :term:`joblib`.
+
+        ``n_jobs`` is an integer, specifying the maximum number of concurrently
+        running workers. If 1 is given, no joblib parallelism is used at all,
+        which is useful for debugging. If set to -1, all CPUs are used. For
+        ``n_jobs`` below -1, (n_cpus + 1 + n_jobs) are used. For example with
+        ``n_jobs=-2``, all CPUs but one are used.
+
+        ``n_jobs`` is ``None`` by default, which means *unset*; it will
+        generally be interpreted as ``n_jobs=1``, unless the current
+        :class:`joblib.Parallel` backend context specifies otherwise.
 
         When ``n_jobs`` is not 1, the estimator being parallelized must be
-        picklable.  This means, for instance, that lambdas cannot be used
-        as estimator parameters.
+        picklable. This means, for instance, that lambda-functions cannot be
+        used as estimator parameters.
```

Comment:
On a side note, maybe it would be good to state that some models based on third party libraries -- such as `pytorch` or `tensorflow` -- might have some pickling issue? I believe we saw a couple of these issues in `joblib` but maybe not so much in `sklearn`?

Reply:
> Actually now that joblib uses the loky backend by default, this restriction no longer applies

Should we just simply remove the comment then? Not sure it's worth mentioning something if it's not a problem by default (especially in the glossary)

Reply:
> Should we just simply remove the comment then?

Alright.